fondant/manifest.py

```diff
@@ -149,6 +149,10 @@ def add_metadata(self, key: str, value: t.Any) -> None:
     def base_path(self) -> str:
         return self.metadata["base_path"]
 
+    @base_path.setter
```

Comment:
I used the `add_metadata()` in my first version but decided against it since I was updating metadata and not adding it. But I guess we could use it instead of the setter.